lib/concurrent/executor/executor_options.rb

```diff
@@ -19,31 +19,41 @@ module ExecutorOptions # :nodoc:
     #
     # @!visibility private
     def get_executor_from(opts = {}) # :nodoc:
-      if (executor = opts[:executor]).is_a? Symbol
+      case
+      when opts.key?(:executor)
         case opts[:executor]
         when :fast
           Concurrent.global_fast_executor
         when :io
           Concurrent.global_io_executor
         when :immediate
-          Concurrent::ImmediateExecutor.new
+          Concurrent.global_immediate_executor
         when :operation
           Kernel.warn '[DEPRECATED] use `executor: :fast` instead'
           Concurrent.global_fast_executor
         when :task
           Kernel.warn '[DEPRECATED] use `executor: :io` instead'
           Concurrent.global_io_executor
+        when Executor
```

Comment:
This is a nifty trick. I didn't realize you could type check in a case statement this way.